shared/middleware/assets_api.rb

```diff
@@ -35,7 +35,16 @@ class AssetsApi < Sinatra::Base
     s3 = Aws::S3::Client.new(region: 'us-east-1')
     s3.list_objects(bucket:CDO.assets_s3_bucket, prefix:prefix).contents.map do |fileinfo|
       filename = %r{#{prefix}/(.+)$}.match(fileinfo.key)[1]
-      {filename:filename, size:fileinfo.size}
+      mime_type = Sinatra::Base.mime_type(filename.split('.').last)
+      case mime_type.split('/').first
+        when 'image'
+          category = 'image'
+        when 'audio'
+          category = 'sound'
+        else
+          halt(500, 'Unexpected mime type: ' + mime_type)
```

Comment:
I've changed the PUT handler to enforce a file type whitelist, so it should no longer be necessary to validate the mime type here. If we did find an invalid mime type here, it would be a server error because the problem is bad data on the server, not any problem with the client's request.